lib/service_base_url_test_kit/service_base_url_validate_group.rb

```diff
@@ -150,15 +162,20 @@ def find_referenced_endpoint(bundle_resource, endpoint_id_ref)
 
         skip_if bundle_resource.entry.empty?, 'The given Bundle does not contain any resources'
 
-        bundle_resource
+        endpoint_list = bundle_resource
           .entry
           .map(&:resource)
           .select { |resource| resource.resourceType == 'Endpoint' }
           .map(&:address)
           .uniq
-          .each do |address|
+
+        check_limit = get_endpoint_availability_limit(endpoint_availability_limit)
+
```

Comment:
I would throw in a message like this so that it is clear that not all endpoints have been checked:

```
        if endpoint_availability_limit.present? && endpoint_availability_limit.to_i < endpoint_list.count 
          info "Only the first #{endpoint_availability_limit.to_i} endpoints of #{endpoint_list.count} total will be checked."
        end
```